`make run-example` panics probabilistically

`make run-example` panics with low probability.
Stack trace:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x1039182]

goroutine 1 [running]:
github.com/pfnet-research/k8s-cluster-simulator/pkg/pod.(*Pod).IsTerminated(0x0, 0x0, 0xed3bc3e96, 0x1fa80c0, 0xc000512bb0)
        /home/hyab/go/src/github.com/pfnet-research/k8s-cluster-simulator/pkg/pod/pod.go:171 +0x22
github.com/pfnet-research/k8s-cluster-simulator/pkg/pod.(*Pod).Delete(0x0, 0x0, 0xed3bc3e96, 0x1fa80c0)
        /home/hyab/go/src/github.com/pfnet-research/k8s-cluster-simulator/pkg/pod/pod.go:192 +0x4d
github.com/pfnet-research/k8s-cluster-simulator/pkg/node.(*Node).DeletePod(0xc000513fb0, 0x0, 0xed3bc3e96, 0x1fa80c0, 0x12cf2f8, 0x7, 0xc000531110, 0x7, 0xf)
        /home/hyab/go/src/github.com/pfnet-research/k8s-cluster-simulator/pkg/node/node.go:118 +0xba
github.com/pfnet-research/k8s-cluster-simulator/pkg/kubesim.(*KubeSim).deletePodFromNode(0xc0003ebf00, 0x12cf2f8, 0x7, 0xc000531110, 0x7)
        /home/hyab/go/src/github.com/pfnet-research/k8s-cluster-simulator/pkg/kubesim/kubesim.go:435 +0x194
github.com/pfnet-research/k8s-cluster-simulator/pkg/kubesim.(*KubeSim).submit.func1(0x12d1ced, 0xb, 0x142aa40, 0xc0005229c0, 0xc0005229c0)
        /home/hyab/go/src/github.com/pfnet-research/k8s-cluster-simulator/pkg/kubesim/kubesim.go:335 +0x915
github.com/pfnet-research/k8s-cluster-simulator/pkg/kubesim/submitter_map.(*SubmitterMap).Range.func1(0x1119900, 0xc00052a0b0, 0x1186960, 0xc0005229c0, 0x0)
        /home/hyab/go/src/github.com/pfnet-research/k8s-cluster-simulator/pkg/kubesim/submitter_map/submitter_map.go:86 +0xaa
sync.(*Map).Range(0xc0005029f0, 0xc000359aa8)
        /usr/local/go/src/sync/map.go:337 +0x136
github.com/pfnet-research/k8s-cluster-simulator/pkg/kubesim/submitter_map.(*SubmitterMap).Range(0xc0005029f0, 0xc000359ae8)
        /home/hyab/go/src/github.com/pfnet-research/k8s-cluster-simulator/pkg/kubesim/submitter_map/submitter_map.go:89 +0x55
github.com/pfnet-research/k8s-cluster-simulator/pkg/kubesim.(*KubeSim).submit(0xc0003ebf00, 0xc0005edc20, 0x8, 0xc000359bc8)
        /home/hyab/go/src/github.com/pfnet-research/k8s-cluster-simulator/pkg/kubesim/kubesim.go:300 +0x70
github.com/pfnet-research/k8s-cluster-simulator/pkg/kubesim.(*KubeSim).Run(0xc0003ebf00, 0x1443c00, 0xc000463100, 0xc0004addc0, 0xc0005229c0)
        /home/hyab/go/src/github.com/pfnet-research/k8s-cluster-simulator/pkg/kubesim/kubesim.go:155 +0x2ad
main.glob..func1(0x1f9b320, 0xc000465940, 0x0, 0x2)
        /home/hyab/go/src/github.com/pfnet-research/k8s-cluster-simulator/example/main.go:81 +0x211
github.com/pfnet-research/k8s-cluster-simulator/vendor/github.com/spf13/cobra.(*Command).execute(0x1f9b320, 0xc00003a0a0, 0x2, 0x2, 0x1f9b320, 0xc00003a0a0)
        /home/hyab/go/src/github.com/pfnet-research/k8s-cluster-simulator/vendor/github.com/spf13/cobra/command.go:766 +0x2cc
github.com/pfnet-research/k8s-cluster-simulator/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x1f9b320, 0x105404f, 0xc0003cff88, 0x407460)
        /home/hyab/go/src/github.com/pfnet-research/k8s-cluster-simulator/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/pfnet-research/k8s-cluster-simulator/vendor/github.com/spf13/cobra.(*Command).Execute(0x1f9b320, 0x6, 0x0)
        /home/hyab/go/src/github.com/pfnet-research/k8s-cluster-simulator/vendor/github.com/spf13/cobra/command.go:800 +0x2b
main.main()
        /home/hyab/go/src/github.com/pfnet-research/k8s-cluster-simulator/example/main.go:36 +0x31
```

--- a/pkg/node/node.go
+++ b/pkg/node/node.go
@@ -115,7 +115,9 @@ func (node *Node) BindPod(clock clock.Clock, v1Pod *v1.Pod) (*pod.Pod, error) {
 func (node *Node) DeletePod(clock clock.Clock, podNamespace, podName string) bool {
 	key := util.PodKeyFromNames(podNamespace, podName)
 	pod, ok := node.pods[key]
-	pod.Delete(clock)
+	if ok {
+		pod.Delete(clock)
+	}
 
 	return ok
 }

